### ISSUE TITLE
Allow installation without relying on local node

### ISSUE THREAD

**Issue by alexislefebvre** [caprover/caprover#2131]
**Is your feature request related to a problem? Please describe.**

When installing Caprover, the step 3 requires to install a NPM package: https://caprover.com/docs/get-started.html#step-3-install-caprover-cli

It was used only to provide a few strings:

```bash
$  caprover serversetup

Setup CapRover machine on your server...

? have you already started CapRover container on your server? Yes
? IP address of your server: 51.[…]
? CapRover server root domain: caprover.[…]
? new CapRover password (min 8 characters): [hidden]
? enter new CapRover password again: [hidden]
? "valid" email address to get certificate and enable HTTPS: […]
? CapRover machine name, with whom the login credentials are stored locally: captain-[…]

CapRover server setup completed: it is available as captain-teth at https://captain.caprover.[…]
```

Then I was able to go to https://captain.caprover.[…] and use it.

---

My goal is to deploy apps from GitLab CI to a remote server, so ideally I wouldn't have to install any package on my local machine. (I should *only* have to install CapRover on the remote server then configure GitLab CI to push code or Docker images)

**Describe the solution you'd like**

These values may be provided as environment variables to the container, because they are known values.

Example:

```bash
docker run -p 80:80 -p 443:443 -p 3000:3000 -e ACCEPTED_TERMS=true \
-e SERVER_ADDRESS=51.[…] -e SERVER_ROOT_DOMAIN=caprover.[…] \
-e CAPROVER_PASSWORD=[…] -e CAPROVER_MACHINE_NAME=captain-[…] \
-e EMAIL_ADDRESS=[…] \
-v /var/run/docker.sock:/var/run/docker.sock -v /captain:/captain caprover/caprover
```

**Describe alternatives you've considered**

The “Best Practices” page shows that we can provide a password: https://caprover.com/docs/best-practices.html#custom-default-password

But someone reading the documentation will reach this part after installing CapRover, or even totally miss it.

**Additional context**

.

**Comment by alexislefebvre** [comment]
By taking the values from the CLI source code https://github.com/caprover/caprover-cli/blob/66a3be524f2bc2e01fc9013a11ab183075cad7aa/src/commands/serversetup.ts#L81

I was able to provide these data to CapRover:

```bash
docker run -p 80:80 -p 443:443 -p 3000:3000 -e ACCEPTED_TERMS=true -e CAPROVER_IP=51.[…] -e CAPROVER_ROOT_DOMAIN=caprover.[…] -e CAPROVER_NEW_PASSWORD=k[…] -e CAPROVER_CERTIFICATE_EMAIL=[…] -e CAPROVER_NAME=caprover-[…] -v /var/run/docker.sock:/var/run/docker.sock -v /var/docker/caprover/captain:/captain caprover/caprover
```

Sadly, CapRover is not reachable through https://captain.caprover.[…], the certificate is self-signed and it only shows “Nothing here yet :/”. So it looks like it starts then fail to finish the install steps.

**Comment by githubsaturn** [comment]
You can set up your instance without the local `caprover` CLI. You can set up your instance on GUI, it'll take just one minute! 

- Install CapRover normally
- Navigate to http://<IP>:3000
- Login using `captain42` - the default password
- Set the root domain on the dashboard
- Re-do login on `HTTP://captain.root.domain.com` - still using the default password
- Click on Enable SSL and enter email address
- Click on FORCE SSL 
- Login again using  `httpS://captain.root.domain.com` - using default password
- Go to settings and change the password

**Comment by alexislefebvre** [comment]
@githubsaturn Thanks for the answer. I tried to test it locally https://caprover.com/docs/run-locally.html#installation but my Docker may be too old to run CapRover.

Anyway should the documentation be updated if the “Step 3” is optional?

**Comment by githubsaturn** [comment]
You don't need to install CapRover on your local machine! You just need to install the CLI on your local machine. CLI is just a client talking to me REST API of CapRover. Was that the misunderstanding?

**Comment by alexislefebvre** [comment]
@githubsaturn When I installed CapRover on the remote server, the port 3000 didn't respond until I used the CLI to run `serversetup`. So I tried to replicate this on a fresh install of CapRover on my local environment, to check if the port 3000 was responding, but it wasn't conclusive.

I will reset CapRover on the remote server and try to install it through the port 3000, without using the CLI.

**Comment by githubsaturn** [comment]
CLI doesn't have any magics. It simply connects to http://IP:3000

**Comment by alexislefebvre** [comment]
I removed Caprover entirely from the server and relaunched it. And I was able to reach it on the port 3000. :rocket:

I think it didn't work previously because I didn't let enough time to CapRover to initialize itself. I stand corrected.

Once I'm connected, I'm facing a confusing message:

> ![image](https://github.com/user-attachments/assets/a505cbe4-4f06-40e4-8911-e5d4b1e9f5cf)

> Use the panel below: This is a non-guided version of the Command Line method. Use this method only for the purpose of experimentation.

If it's only for experimentation, then it's not officially recommended?
